src/OpenIddict.Client/OpenIddictClientBuilder.cs

```diff
@@ -200,67 +200,84 @@ public OpenIddictClientBuilder AddDevelopmentEncryptionCertificate(X500Distingui
             throw new ArgumentNullException(nameof(subject));
         }
 
-        using var store = new X509Store(StoreName.My, StoreLocation.CurrentUser);
-        store.Open(OpenFlags.ReadWrite);
-
-        // Try to retrieve the existing development certificates from the specified store.
-        // If no valid existing certificate was found, create a new encryption certificate.
-        var certificates = store.Certificates.Find(X509FindType.FindBySubjectDistinguishedName, subject.Name, validOnly: false)
-            .OfType<X509Certificate2>()
-            .ToList();
-
-        if (!certificates.Exists(static certificate => certificate.NotBefore < DateTime.Now && certificate.NotAfter > DateTime.Now))
+        Services.AddOptions<OpenIddictClientOptions>().Configure<IServiceProvider>((options, serviceProvider) =>
         {
-#if SUPPORTS_CERTIFICATE_GENERATION
-            using var algorithm = OpenIddictHelpers.CreateRsaKey(size: 2048);
+#if SUPPORTS_TIME_PROVIDER
+            var timeProvider = options.TimeProvider ?? serviceProvider.GetService<TimeProvider>();
+            var notBefore = timeProvider?.GetLocalNow() ?? DateTimeOffset.Now;
```

Comment:
Since this variable is used for both `NotAfter` and not `NotBefore` checks, maybe a less specific name - like `now`, that you used in other files - would be better?